azure-active-directory-b2c/README.adoc

```diff
@@ -0,0 +1,392 @@
+            authorization-grant-type: authorization_code
+            redirect-uri: "{baseUrl}/login/oauth2/code/"
+            # Fill in `${client-id}` from `Application ID`
+            # We can use ${client-id} to acquire access token that can be used against your own service or web API.
+            # Related doc: https://docs.microsoft.com/en-us/azure/active-directory-b2c/access-tokens#openid-connect-scopes
+            scope: ${client-id}, openid, offline_access, profile
+
+# this uri is used to clear your cache when you logout
+logout-success-url: https://sampleTenant1.b2clogin.com/sampleTenant1.onmicrosoft.com/B2C_1_signupsignin1/oauth2/v2.0/logout?post_logout_redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Flogin
+```
+
+Next step, we need to fill these placeholders in application.yml: `${your-token-uri}`, `${client-id}` etc.
+
+==== 1.2. Prepare Azure Active Directory B2C resources.
+
+===== 1.2.1. Create your Azure Active Directory B2C tenant.
+
+Follow the https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-tenant[AAD B2C tenant creation] guide, create a tenant named "sample-tenant-1" and configure initial domain name with "sampleTenant1".
+
+===== 1.2.2. Register your Azure Active Directory B2C application.
+
+Follow the https://docs.microsoft.com/azure/active-directory-b2c/tutorial-register-applications[AAD B2C application registry] guide, create a web application named "client-1" and create a client secret.
+Please ensure that your b2c application's `Redirect URL` is configured to `http://localhost:8080/login/oauth2/code/`.
+In order to jump to the correct page (home) after `profile-edit` and `password-reset`, we need to add `http://localhost:8080/` in the redirect uri too.
+You can add additional `Redirect URL` with https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app#add-a-redirect-uri[Add Redirect URL].
+
+===== 1.2.3. Create user flows.
+
+Follow the https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-user-flows[AAD B2C user flows creation] guide, create a sign-up and sign-in user flow named "signupsignin1".
+
+===== 1.2.4. Enable forgot password.
+
+Follow the guide of https://docs.microsoft.com/azure/active-directory-b2c/add-password-reset-policy?pivots=b2c-user-flow#self-service-password-reset-recommended[AAD B2C enable forgot password].
+
+===== 1.2.5. Enable profile edit.
+
+Follow the https://docs.microsoft.com/azure/active-directory-b2c/add-profile-editing-policy?pivots=b2c-user-flow[AAD B2C set up a profile edit flow] guide, create a Profile editing flow named "profileediting".
+
+===== 1.2.6. Enable password reset.
+
+Follow the https://docs.microsoft.com/azure/active-directory-b2c/add-password-reset-policy?pivots=b2c-user-flow#create-a-password-reset-user-flow[AAD B2C set up a password reset user flow] guide, create a password reset user flow named "passwordreset".
+
+==== 1.3. Fill necessary values in sample project.
```

Comment:
How about filling these values when creating related resources?

Just like this: https://github.com/chenrujun/azure-spring-boot-samples/tree/bb72202e47d03ef47227d324b1fb0fa084d78ae7/azure-active-directory#121-get-an-azure-active-directory-tenant